controller/test/alarm/alarm_test.cpp

```diff
@@ -12,18 +12,50 @@ class AlarmTest : public testing::Test {
 
 TEST_F(AlarmTest, NotAvailableAfterInit) { ASSERT_FALSE(alarm_available()); }
 
-TEST_F(AlarmTest, AddOneAlarm) {
+TEST_F(AlarmTest, AddAndRemoveOneAlarm) {
   char alarm_data[ALARM_DATALEN];
   memset(alarm_data, 'x', sizeof(alarm_data));
 
   alarm_add(dataID::alarm_1, alarm_data);
   ASSERT_TRUE(alarm_available());
+  alarm_remove();
+  ASSERT_FALSE(alarm_available());
+}
+
+TEST_F(AlarmTest, OverflowAlarmQueue) {
+  for (uint8_t i = 0; i <= ALARM_NODES; i++) {
+    char alarm_data[ALARM_DATALEN];
+    memset(alarm_data, 'x', sizeof(alarm_data));
+    alarm_add(dataID::alarm_1, alarm_data);
+  }
+  ASSERT_TRUE(alarm_available());
+  // TODO(SarahAdelaide) when logging is implemented, validate log is generated
+}
+
+TEST_F(AlarmTest, AddAndReadAlarm) {
+  while (alarm_available()) {
+    alarm_remove();
+  }
+
+  char alarm_data_a[ALARM_DATALEN];
+  memset(alarm_data_a, 'y', sizeof(alarm_data_a));
+  dataID alarm_id_a = dataID::alarm_2;
+  alarm_add(alarm_id_a, alarm_data_a);
+  // checking that alarm data got copied, not referenced
+  memset(alarm_data_a, 'x', sizeof(alarm_data_a));
+  alarm_id_a = dataID::alarm_1;
+
+  enum dataID alarm_id_b;
+  uint32_t alarm_timestamp_b;
+  char alarm_data_b[ALARM_DATALEN];
+  int32_t res = alarm_read(&alarm_id_b, &alarm_timestamp_b, alarm_data_b);
```

Comment:
Drive-by comment: For functions that return errors, it'd be cool to use https://en.cppreference.com/w/cpp/language/attributes/nodiscard so that callers are required by the compiler to check the error.

Reply:
Thanks, Eugene! Maybe I'm misunderstanding the purpose of nodiscard, but it looks to me like it would be most usefully deployed in the codebase, not the test suites? Is this what you're thinking?

Reply:
Yes, I meant alarm_read itself (and other functions that may return an error), not its usage in this test.